### PR TITLE
testOnlyDebug command and changes in runMain and launchTask

### PR DESCRIPTION
**Pull request by vovapolu**

### CODE REVIEW

**Comment by fommil** [comment]
@dwijnand FYI

**Comment by dwijnand** [comment]
Give thanks, men, to Leonidas and the brave #300! TO VICTORY!

**Comment by fommil** [comment]
@vovapolu you could put some code in the sbt package, and expose what you need, no reflection needed

**Comment by fommil** [comment]
btw, the tests failed on this due to a compiler warning (escalated to error in CI) and you haven't signed the CLA

File: src/main/scala/SbtHelper.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// for using some private[sbt] functionality
+package sbt
+
+import sbt.ConcurrentRestrictions.Tag
+import Tests.{Output => TestOutput, _}
+import sbt.testing.Runner
+
+object SbtHelper {
+  def consturctForkTests(runners: Map[TestFramework, Runner], tests: List[TestDefinition], config: Execution, classpath: Seq[File], fork: ForkOptions, log: Logger, tag: Tag): Task[TestOutput] = {
```
Comment (fommil) [inline src/main/scala/SbtHelper.scala]: typo

File: src/sbt-test/sbt-ensime/ensime-run-main/test [hunk 2]
```diff
@@ -1,26 +1,32 @@
...
-$ must-mirror args2 output_args2
-> ensime-run-main/ensimeRunMain -Dtesting_key3=value3 -Xms2G -Xmx2G runEnsimeMain.printArgs output_args3
-$ must-mirror args3 output_args3
-> ensime-run-main/ensimeRunMain testing_key1=value1 testing_key2=value2 runEnsimeMain.printArgs output_args4
-$ must-mirror args4 output_args4
-> ensime-run-main/ensimeRunMain testing_default_key2=something -Dtesting_default_key1=something runEnsimeMain.printArgs output_args5
-$ must-mirror args5 output_args5
-> ensime-run-main/ensimeRunMain testing_key1=value1 testing_key2=value2 testing_default_key2=something -Dtesting_default_key1=something -Dtesting_key3=value3 -Xms2G -Xmx2G runEnsimeMain.printArgs output_args6 -arg
-$ must-mirror args6 output_args6
+#> ensime-run-main/compile
```
Comment (fommil) [inline src/sbt-test/sbt-ensime/ensime-run-main/test]: lots of commented out tests?

**Comment by fommil** [comment]
sweet, good to see it alive again :smile: I'll look tomorrow... brain fried from day at work

**Comment by fommil** [comment]
we should remove the `debugging` mutating hack... I think it's in `EnsimePlugin.scala` if it's not in extras. This completes supercedes it.

**Review by fommil — APPROVED** [review]
ok, looks good! Let's remove the old debugging mechanism, merge this, and release with updated docs.

File: src/sbt-test/sbt-ensime/bad-build-scala/test [hunk 3]
```diff
@@ -2,7 +2,7 @@
 > ensimeConfigProject
 > ensimeExpect project/.ensime ensimeConfigProject.expect
 > checkJavaOptions
-> debugging
-> checkJavaOptions -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005
-> debuggingOff
-> checkJavaOptions
+#> debugging
```
Comment (fommil) [inline src/sbt-test/sbt-ensime/bad-build-scala/test]: Remove instead of comment?

**Review by fommil — CHANGES_REQUESTED** [review]
great stuff, I think we can merge this once you remove the tests instead of just commenting them.

Would you like to practice your English by updating the docs? http://ensime.org/build_tools/sbt/#debugging-example If not, I will happily update them :smile:

**Review by fommil — APPROVED** [review]

**Comment by fommil** [comment]
I'll cut a new release tomorrow, and announce it when you've updated the docs. This will make a lot of people happy!

**Pull request merged**